src/Hedgehog/PropertyResult.fs

```diff
@@ -0,0 +1,103 @@
+namespace Hedgehog
+
+/// Internal discriminated union representing either a synchronous or asynchronous property result
+[<RequireQualifiedAccess>]
+type internal PropertyResult<'a> =
+    | Sync of Journal * Outcome<'a>
+    | Async of Async<Journal * Outcome<'a>>
+
+[<RequireQualifiedAccess>]
+module internal PropertyResult =
+
+    /// Create a synchronous PropertyResult from a journal and outcome
+    let ofSync (journal : Journal) (outcome : Outcome<'a>) : PropertyResult<'a> =
+        PropertyResult.Sync (journal, outcome)
+
+    /// Create an async PropertyResult from an async computation that produces a journal and outcome
+    let ofAsync (asyncResult : Async<Journal * Outcome<'a>>) : PropertyResult<'a> =
+        PropertyResult.Async asyncResult
+
+    /// Create an async PropertyResult by capturing exceptions from an async computation
+    let ofAsyncWith (asyncComputation : Async<'a>) : PropertyResult<'a> =
+        PropertyResult.Async (async {
```

Comment:
This catches all exceptions and converts them to failures. This is correct, but perhaps we should consider whether we want to catch `OperationCanceledException` separately (like what's done in `ofTask`) to provide a more specific message.

Reply:
Good catch! 👍